geo/src/algorithm/contains.rs

```diff
@@ -340,11 +340,29 @@ where
     T: CoordinateType,
 {
     fn contains(&self, coord: &Coordinate<T>) -> bool {
-        let sign_1 = utils::sign(coord, &self.0, &self.1);
-        let sign_2 = utils::sign(coord, &self.1, &self.2);
-        let sign_3 = utils::sign(coord, &self.2, &self.0);
+        let s = self.0.y * self.2.x - self.0.x * self.2.y
+            + (self.2.y - self.0.y) * coord.x
+            + (self.0.x - self.2.x) * coord.y;
+        let t = self.0.x * self.1.y - self.0.y * self.1.x
+            + (self.0.y - self.1.y) * coord.x
+            + (self.1.x - self.0.x) * coord.y;
+
+        if (s < T::zero()) != (t < T::zero()) {
+            return false;
+        }
+
+        let a = (T::zero() - self.1.y) * self.2.x
+            + self.0.y * (self.2.x - self.1.x)
+            + self.0.x * (self.1.y - self.2.y)
+            + self.1.x * self.2.y;
 
-        (sign_1 == sign_2) && (sign_2 == sign_3)
+        if a == T::zero() {
```

Comment:
I haven't analyzed the math of the solution, but only looking at the transcription...

what's this `a==T::zero()` branch about? It's not present in the linked example: https://stackoverflow.com/questions/2049582/how-to-determine-if-a-point-is-in-a-2d-triangle/20861130#20861130

Reply:
My initial thinking: the area of a degenerate triangle (formed by collinear points) is zero, therefor a point is never contained by a degenerate triangle (return `false`). However, reading the suggestion from #473, checking if the given point contains any of the triangle points, I realize that makes more sense. How do you suggest to solve this?

Reply:
I think this makes sense, and the unit test verifies the behavior. Thanks!